resources/sql_scripts/update_scripts/firmware_max_retry_limit_update.sql

```diff
@@ -0,0 +1,23 @@
+-- Run this SQL update script if you already have a deployed CV Manager PostgreSQL database
+
+CREATE TABLE IF NOT EXISTS public.consecutive_firmware_upgrade_failures
+(
+   rsu_id integer NOT NULL,
+   consecutive_failures integer NOT NULL,
+   CONSTRAINT consecutive_firmware_upgrade_failures_pkey PRIMARY KEY (rsu_id),
+   CONSTRAINT fk_rsu_id FOREIGN KEY (rsu_id)
+      REFERENCES public.rsus (rsu_id) MATCH SIMPLE
+      ON UPDATE NO ACTION
+      ON DELETE NO ACTION
+);
+
+CREATE TABLE IF NOT EXISTS public.max_retry_limit_reached_instances
+(
+   rsu_id integer NOT NULL,
+   reached_at timestamp without time zone NOT NULL,
+   CONSTRAINT max_retry_limit_reached_instances_pkey PRIMARY KEY (rsu_id, reached_at),
+   CONSTRAINT fk_rsu_id FOREIGN KEY (rsu_id)
+      REFERENCES public.rsus (rsu_id) MATCH SIMPLE
+      ON UPDATE NO ACTION
+      ON DELETE NO ACTION
+);
```

Comment:
question/suggestion (non-blocking): could these two tables be combined into one like the following? I'm unsure about the context and the database design standards, but combining the two would reduce data duplication and (possibly) make the debugging developer's life a little easier since they wouldn't need to query multiple tables for the information.

```suggestion
CREATE TABLE IF NOT EXISTS public.consecutive_firmware_upgrade_failures
(
   rsu_id integer NOT NULL,
   consecutive_failures integer NOT NULL,
   last_update_attempted_at timestamp without time zone NOT NULL,
   CONSTRAINT consecutive_firmware_upgrade_failures_pkey PRIMARY KEY (rsu_id),
   CONSTRAINT fk_rsu_id FOREIGN KEY (rsu_id)
      REFERENCES public.rsus (rsu_id) MATCH SIMPLE
      ON UPDATE NO ACTION
      ON DELETE NO ACTION
);
```

The table could be used in the code in a similar way as it is currently. The queries would just need to check `consecutive_failures` against the configured `FW_UPGRADE_MAX_RETRY_LIMIT`. Obviously, other code changes would be needed, but this is the most obvious.

Reply:
One table tracks consecutive failure counts to help decide whether to continue upgrade attempts. The second table, `public.max_retry_limit_reached_instances`, serves as a log for instances where the maximum retry limit has been reached. Developers can reference this log for additional context. I don't think combining these two tables would preserve their distinct purposes.